fix(stubs): teleport stub children as a function

cexbrayat opened the pull request:
Fixes #1829

I'm not super convinced by this fix, but it indeed looks like @freakzlike intuition is right: we need to return the children as a function for Teleport to avoid the warning.

Maybe @xanf you have a better idea on how to handle that?

freakzlike commented on src/vnodeTransformers/util.ts:
```diff
@@ -53,8 +53,12 @@ export const createVNodeTransformer = ({
       registerStub({ source: originalType, stub: transformedType })
+      // https://github.com/vuejs/test-utils/issues/1829
+      // Teleport should return child nodes as a function
+      if (transformedType.name === 'teleport') {
```
It might be better to check for the `Teleport` component instead of the name
```suggestion
      if (originalType === Teleport) {
```

cexbrayat replied:
👍 but sadly this does not compile: I ended up borrowing the check from core: `const isTeleport = (type: any): boolean => type.__isTeleport`

freakzlike commented on src/vnodeTransformers/util.ts:
```diff
@@ -31,15 +31,15 @@ export const createVNodeTransformer = ({
     const cachedTransformation = transformationCache.get(originalType)
     if (cachedTransformation) {
-      return [cachedTransformation, ...restVNodeArgs]
+      return [cachedTransformation, props, children, ...restVNodeArgs]
```
Do we need to make this also as function for teleport?

freakzlike reviewed (commented):
I have no better solution for this.
Can you add a spy on `console.warn` on one of the teleport tests?

cexbrayat commented:
@freakzlike I added a test to check that the warning is not logged anymore, and added a check in other tests when it _is_ logged, but when that we expect it.

Maybe we should do the same for the warnings logged by vue compact (either mock them or fix them if needed) to avoid polluting the test logs, and missing a regression like this one in the future

Anyway, this should fix the issue, unless someone comes up with a better idea. I'll let you give a last check and merge
